rust/connlib/tunnel/src/client.rs

```diff
@@ -54,36 +60,37 @@ where
     ///
     /// Once added, when a packet for the resource is intercepted a new data channel will be created
     /// and packets will be wrapped with wireguard and sent through it.
-    pub fn add_resource(
+    pub fn add_resources(
         &mut self,
-        resource_description: ResourceDescription,
+        resources: &[ResourceDescription],
     ) -> connlib_shared::Result<()> {
-        if let Some(resource) = self.role_state.resource_ids.get(&resource_description.id()) {
-            if resource.has_different_address(resource) {
-                self.remove_resource(resource.id());
+        for resource_description in resources {
+            if let Some(resource) = self.role_state.resource_ids.get(&resource_description.id()) {
+                if resource.has_different_address(resource) {
```

Comment:
Would we hit this today? I thought resource addresses can't be changed?

Reply:
The message from the portal is `ResourceCreatedOrUpdated` I'm not sure what are the situations in the portal that can trigger that but anyways it's better to have it here already.